---
title: "A tour of MethylBioPaths"
author: "Juan De Los Rios"
date: "`r format(Sys.time(), '%d %b %Y')`"
output: 
  rmarkdown::html_vignette:
    toc: true
    number_sections: false
vignette: >
  %\VignetteIndexEntry{A tour of MethylBioPaths}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
library(knitr)
opts_chunk$set(fig.align = "center", 
               out.width = "90%",
               fig.width = 6, fig.height = 5.5,
               dev.args=list(pointsize=10),
               par = TRUE, # needed for setting hook 
               collapse = TRUE, # collapse input & ouput code in chunks
               warning = FALSE)

knit_hooks$set(par = function(before, options, envir)
  { if(before && options$fig.show != "none") 
       par(family = "sans", mar=c(4.1,4.1,1.1,1.1), mgp=c(3,1,0), tcl=-0.5)
})
set.seed(1) # for exact reproducibility
```

## Introduction

**MethylBioPaths** is an R package for analyzing DNA methylation data, specifically the biological pathways that genes involved in driving differential methylation are involved in and the relationship between mutation types and differential methylation. **This document gives a tour of MethylBiopPaths (version 0.1.0)**. It was written in R Markdown, using the [knitr](https://cran.r-project.org/package=knitr) package for production.

To download **MethylBioPaths**, use the following commands:

``` r
require("devtools")
install_github("Juan-De-Los-Rios/MethylBioPaths", build_vignettes = TRUE)
library("MethylBioPaths")
```
To list all sample functions available in the package:
``` r
ls("package:MethylBioPaths")
```
To list all sample datasets available in the package:
``` r
data(package = "MethylBioPaths")
```

<br>

## Functions

MethylBioPaths contains two functions for analyzing DNA methylation data. This tutorial will guide you through how to use these functions and how to interpret their results. The tutorial will assume you will be using the following example datasets to test this package's functions.

```r
library(MethylMix)
data(METcancer)
data(METnormal)
data(GEcancer)
data(mutationExample)
```

**methylMutationAssociation** A function that generates a plot highlighting the relationship between mutation category and differential DNA methylation. It accepts 4 datasets as inputs: a matrix containing DNA methylation data from a diseased sample, a matrix containing DNA methylation data from a baseline sample, a matrix containing gene expression data from a diseased sample, and a list categorizing genes present in each sample based on mutations they possess.

```r
methylMutationAssociation(METcancer, METnormal, GEcancer, mutationExample)
#Produces box plot illustrating relationship between differential methylation and mutation category
```

The resulting boxplot demonstrates the range of differential methylation values each mutation category encompasses through the driver genes belonging to each category, thus visualizing the ultimate impact each type of mutation has on DNA methylation in the disease of interest.

**methylOverrep** A function that generates a plot summarizing the findings of an overrepresentation analysis performed using the driver genes derived from DNA methylation samples. It accepts 3 datasets as inputs: a matrix containing DNA methylation data from a diseased sample, a matrix containing DNA methylation data from a baseline sample, and a matrix containing gene expression data from a diseased sample. It also accepts two integers, pValue and qValue, representing the cutoffs for how stringent you want to significance threshold of the overrepresentation analysis to be.

```r
resPlot <- methylOverrep(METcancer, METnormal, GEcancer, 1, 1)
resPlot
#Returns bar plot summarizing results of biological pathway analysis
```

The resulting barplot summarizes the results of an overrepresentation analysis of the genes driving the differential methylation in the input samples. Each bar corresponds to a category of biological process and the plot as a whole presents the ten most enriched categories as well as a comparison of the number of genes that belonged to each categroy.

<br>

## References

Carlson M (2022). org.Hs.eg.db: Genome wide annotation for Human_. R package version 3.15.0.

Chang W, Cheng J, Allaire J, Sievert C, Schloerke B, Xie Y, Allen J, McPherson J, Dipert A, Borges B (2022). _shiny: Web Application Framework for R_. R package version 1.7.3, <https://CRAN.R-project.org/package=shiny>.

Gevaert O (2022). MethylMix: Identifying methylation driven cancer genes. R package version 2.26.0. 

R Core Team (2022). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL https://www.R-project.org/

T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141

----

```{r}
sessionInfo()
```
